src/core/config/into_document.rs

```diff
@@ -214,19 +265,22 @@ fn config_document(config: &ConfigModule) -> ServiceDocument {
         })));
     }
     for (name, union) in config.unions.iter() {
-        definitions.push(TypeSystemDefinition::Type(pos(TypeDefinition {
-            extend: false,
-            description: None,
-            name: pos(Name::new(name)),
-            directives: Vec::new(),
-            kind: TypeKind::Union(UnionType {
-                members: union
-                    .types
-                    .iter()
-                    .map(|name| pos(Name::new(name.clone())))
-                    .collect(),
-            }),
-        })));
+        // unions are only supported as the output types
+        if !config.input_types.contains(name) {
+            definitions.push(TypeSystemDefinition::Type(pos(TypeDefinition {
+                extend: false,
+                description: None,
+                name: pos(Name::new(name)),
+                directives: Vec::new(),
+                kind: TypeKind::Union(UnionType {
+                    members: union
+                        .types
+                        .iter()
+                        .map(|name| pos(Name::new(name.clone())))
+                        .collect(),
+                }),
+            })));
+        }
```

Comment:
I'm struggling in doing this. I've put my efforts into the pr https://github.com/tailcallhq/tailcall/pull/2011/files but it doesn't work yet. I tried to add yaml config in execution_spec and get working merged graphql config

The issue I'm facing:
- I've added conversion for union types only `into_document` but unexpectedly for me it's not enough since similar conversion is required when doing config -> blueprint. I'm not sure it's a good approach to duplicate it but I don't see a right path to reuse it.
- `resolve_ambigiuous_types` is only called on generator currently and should be duplicated for tests and cli or again reused on some level but I'm not sure where.

After spending some time on trying to solve it in some quick way I really in need for an advice how to better handle it.

Reply:
it's solved by changing the implementation